src/linear_algebra/basic.lean

```diff
@@ -2004,6 +2011,23 @@ end linear_equiv
 
 namespace submodule
 
+section semimodule
+
+variables [semiring R] [add_comm_monoid M] [semimodule R M]
+
+/-- If `s ≤ t`, then we can view `s` as a submodule of `t` by taking the comap
+of `t.subtype`. -/
+def comap_subtype_equiv_of_le {p q : submodule R M} (hpq : p ≤ q) :
+p.comap q.subtype ≃ₗ[R] p :=
```

Comment:
I find this statement very  difficult to read. I think this is a bad use of dot notation, it really feels like `p` is pulling back `q.subtype` instead of the other way around (and even the docstring is suggesting that!). The same problem happens with `filter.(co)map` and always try to avoid it.
I would `open submodule` and then write `comap q.subtype p ≃ₗ[R] p`. I won't insist on this if you really like your version.

Reply:
I don't mind either way, so I'm happy to change it. Should I also change the lines starting from `linear_algebra/basic.lean:1362` (in a different PR)?

```lean
@[simp] lemma comap_subtype_eq_top {p p' : submodule R M} :
  p'.comap p.subtype = ⊤ ↔ p ≤ p' :=
eq_top_iff.trans $ map_le_iff_le_comap.symm.trans $ by rw [map_subtype_top]

@[simp] lemma comap_subtype_self : p.comap p.subtype = ⊤ :=
comap_subtype_eq_top.2 (le_refl _)
```